include/oneapi/dpl/pstl/parallel_backend.h

```diff
@@ -15,42 +15,41 @@
 #ifndef _ONEDPL_PARALLEL_BACKEND_H
 #define _ONEDPL_PARALLEL_BACKEND_H
 #include "onedpl_config.h"
+
+// Select a parallel backend
+#if ONEDPL_USE_TBB_BACKEND || (!defined(ONEDPL_USE_TBB_BACKEND) && !ONEDPL_USE_OPENMP_BACKEND && _ONEDPL_TBB_AVAILABLE)
+#    include "parallel_backend_tbb.h"
+#    define _ONEDPL_PAR_BACKEND_TBB 1
+#elif ONEDPL_USE_OPENMP_BACKEND || (!defined(ONEDPL_USE_OPENMP_BACKEND) && _ONEDPL_OPENMP_AVAILABLE)
+#    include "parallel_backend_omp.h"
+#    define _ONEDPL_PAR_BACKEND_OPENMP 1
+#else
+#    include "parallel_backend_serial.h"
+#    define _ONEDPL_PAR_BACKEND_SERIAL 1
+#endif
+
 #if _ONEDPL_BACKEND_SYCL
 #    include "hetero/dpcpp/parallel_backend_sycl.h"
 #    if _ONEDPL_FPGA_DEVICE
 #        include "hetero/dpcpp/parallel_backend_sycl_fpga.h"
 #    endif
 #endif
 
-#if _ONEDPL_PAR_BACKEND_TBB
-#    include "parallel_backend_tbb.h"
 namespace oneapi
 {
 namespace dpl
 {
-namespace __par_backend = __tbb_backend;
-}
-} // namespace oneapi
+namespace __par_backend =
+#if _ONEDPL_PAR_BACKEND_TBB
+    __tbb_backend;
 #elif _ONEDPL_PAR_BACKEND_OPENMP
-#    include "parallel_backend_omp.h"
-namespace oneapi
-{
-namespace dpl
-{
-namespace __par_backend = __omp_backend;
-}
-} // namespace oneapi
+    __omp_backend;
 #elif _ONEDPL_PAR_BACKEND_SERIAL
-#    include "parallel_backend_serial.h"
-namespace oneapi
-{
-namespace dpl
-{
-namespace __par_backend = __serial_backend;
-}
-} // namespace oneapi
+    __serial_backend;
 #else
-_PSTL_PRAGMA_MESSAGE("Parallel backend was not specified");
+#    error "Parallel backend was not specified"
```

Comment:
It still doesn't look accurate to me because of assignment.

How about rewriting it to 

```
#if _ONEDPL_PAR_BACKEND_TBB
    namespace __par_backend = __tbb_backend;
#elif _ONEDPL_PAR_BACKEND_OPENMP
    namespace __par_backend = __omp_backend;
#else
    namespace __par_backend = __serial_backend;
#    if !_ONEDPL_PAR_BACKEND_SERIAL
#        error "whatever"
#    endif // !_ONEDPL_PAR_BACKEND_SERIAL
#endif
```

At least I don't have a problem with repeating ``namespace __par_backend`` in each branch. To me it's more readable

Reply:
The only problem is we have two errors in the error log instead of one. If we want to completely avoid the issue with ``__serial_backend`` being not defined, then I agree with you.

For me either way is fine: I mean current implementation or the option that I am proposing.

So, it's up to you to decide. Approved, by the way.

Reply:
Update: I've checked two implementations (current one and the one suggested by myself) and in current one we have even more errors. So, I see no problem with what I suggest because preprocessor time error would still notify the user. If we want to completely avoid this error we can do #if-#else in the outer #else branch.

Reply:
That looks perfectly fine to me.